charts/linkerd2-cni/templates/cni-plugin.yaml

```diff
@@ -56,6 +56,7 @@ metadata:
   namespace: {{.Values.namespace}}
   labels:
     {{.Values.cniResourceLabel}}: "true"
+{{ include "partials.image-pull-secrets" . }}
```

Comment:
`charts/linkerd2-cni` is a separate chart that doesn't receive the global config from `charts/linkerd2/values.yml`. In order to use `global.imagePullSecrets` in this chart, you need to declare it in `charts/linkerd2-cni/values.yaml`.